ln-gateway/src/lib.rs

```diff
@@ -157,21 +157,13 @@ async fn background_fetch(federation_client: Arc<GatewayClient>, _ln_client: Arc
     // TODO: also try to drive forward payments that were interrupted
     loop {
         let least_wait_until = Instant::now() + Duration::from_millis(100);
-        let pending_fetches = federation_client
-            .list_fetchable_coins()
-            .into_iter()
-            .map(|out_point| {
-                // TODO: get rid of cloning
-                let federation_client = federation_client.clone();
-                async move {
-                    if let Err(e) = federation_client.fetch_coins(out_point).await {
-                        debug!(error = %e, "Fetching coins failed");
-                    }
-                }
-            })
-            .collect::<Vec<_>>();
-        futures::future::join_all(pending_fetches).await;
-        minimint_api::task::sleep_until(least_wait_until).await;
+        match federation_client.fetch_all_coins().await {
+            Ok(o) if o.is_empty() => {
+                minimint_api::task::sleep_until(least_wait_until).await;
+            }
+            Err(e) => debug!(error = %e, "Fetching coins failed"),
```

Comment:
we should also sleep in error case, otherwise it spams the server with a lot of request.